xarray/core/common.py

```diff
@@ -435,3 +517,64 @@ def _possibly_convert_objects(values):
     datetime64 and timedelta64, according to the pandas convention.
     """
     return np.asarray(pd.Series(values.ravel())).reshape(values.shape)
+
+
+def _get_fill_value(dtype):
+    """Return a fill value that appropriately promotes types when used with
+    np.concatenate
+    """
+    _, fill_value = _maybe_promote(dtype)
+    return fill_value
+
+
+def _empty_like_dataarray(arr, keep_attrs=False, fill_value=None):
+    """empty DataArray"""
+    from .dataarray import DataArray
+
+    attrs = arr.attrs if keep_attrs else {}
+
+    if fill_value is None:
+        values = np.empty_like(arr)
+    elif fill_value is True:
+        dtype, fill_value = _maybe_promote(arr.dtype)
+        values = np.full_like(arr, fill_value=fill_value, dtype=dtype)
+    else:
+        dtype, _ = _maybe_promote(np.array(fill_value).dtype)
+        values = np.full_like(arr, fill_value=fill_value, dtype=dtype)
+
+    return DataArray(values, dims=arr.dims, coords=arr.coords, attrs=attrs)
+
+
+def empty_like(xray_obj, keep_attrs=False, fill_value=None):
+    """Return a new object with the same shape and type as a given object.
+
+    Parameters
+    ----------
+    xray_obj : DataArray or Dataset
+        Return a full object with the same shape/dims/coords/attrs.
+            `func` is calculated over all dimension for each group item.
+    keep_attrs : bool, optional
+        If True, the datasets's attributes (`attrs`) will be copied from
+        the original object to the new one.  If False (default), the new
+        object will be returned without attributes.
+    fill_value : scalar, optional
```

Comment:
This seems like a weird argument to have here -- shouldn't this always be inferred on empty_like? Yes, this would make sense on full_like.